### ISSUE TITLE
New Rule Idea: Property var-pattern(s) should appear at the end of is-pattern

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#8195]
If you are using an is-pattern that deconstructs the target, you can use a lot of patterns:

- `thing is { Prop1: true }` (literal)
- `thing is { Prop1: > 10 }` (constraint)
- `thing is { Prop1: var p }` (var-pattern)

When the `var` pattern is mixed with other patterns, _all_ the `var` patterns should be at the end of the expression.

### Noncompliant Code
```csharp
if (thingy is { Prop1: var p1, Count: > 10, Prop2: string p2, Flag: true })
{
    // work
}
```

### Compiant Code
```csharp
if (thingy is { Count: > 10, Flag: true, Prop1: var p1, Prop2: string p2 })
{
    // work
}
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1661](https://sonarsource.atlassian.net/browse/NET-1661)

[NET-1661]: https://sonarsource.atlassian.net/browse/NET-1661?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
